db/migrate/20170427152006_remove_invalid_hawkular_endpoints.rb

```diff
@@ -0,0 +1,83 @@
+class RemoveInvalidHawkularEndpoints < ActiveRecord::Migration[5.0]
+  class ExtManagementSystem < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class Endpoint < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class Authentication < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    ems_container_ids = ExtManagementSystem.where(
+      :type => %w(ManageIQ::Providers::Openshift::ContainerManager ManageIQ::Providers::Kubernetes::ContainerManager)
+    ).pluck(:id)
+    delete_ids = Endpoint.where(
+      :resource_type => 'ExtManagementSystem',
+      :resource_id   => ems_container_ids,
+      :role          => "hawkular",
+      :hostname      => ["", nil],
+    ).pluck(:resource_id)
+
+    unless delete_ids.empty?
+      say_with_time("Removing invalid endpoint and authentication for ems ids [#{delete_ids}]") do
+        Endpoint.where(
+          :resource_type => 'ExtManagementSystem',
+          :resource_id   => delete_ids,
+          :role          => "hawkular",
+        ).destroy_all
+        Authentication.where(
+          :authtype      => "hawkular",
+          :resource_type => 'ExtManagementSystem',
+          :resource_id   => delete_ids
+        ).destroy_all
+      end
+    end
+  end
+
+  def down
+    ems_containers_by_id = ExtManagementSystem.where(
+      :type => %w(ManageIQ::Providers::Openshift::ContainerManager ManageIQ::Providers::Kubernetes::ContainerManager)
+    ).map { |ems| [ems.id, ems] }.to_h
```

Comment:
Minor, but `.map { |ems| [ems.id, ems] }.to_h` can be shortened to `.index_by(&:id)`